### PR TITLE
Add customized 62 key layout called "nobreakfast/nb62"

### PR DESCRIPTION
**Pull request by Nobreakfast**
Add customized 62 key layout called "nobreakfast/nb62"

I add new customized 62 key layout called `nobreakfast/nb62` in folder `qmk_firmware/keyboards/` with userspace `nobreakfast`

## Description

Layout: nobreakfast/nb62

The main difference of this `nb62` layout and general `GH60` is that split the space key into two parts (2.25+2.75). Besides, the left space is aligned with the "C" key, and right space is aligned with the "M" key, which are same for Mac users.

Add file and folder:
nobreakfast
└── nb62
    ├── Makefile
    ├── config.h
    ├── keymaps
        └── default
           └── keymap.c
    ├── nb62.c
    ├── nb62.h
    └── rules.mk

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/nobreakfast/nb62/rules.mk [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+BOOTMAGIC_ENABLE = yes	# Virtual DIP switch configuration(+1000)
+LTO_ENABLE = yes
+RGBLIGHT_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+
+# disable
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+NKRO_ENABLE = no		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+VIA_ENABLE = no
```
Comment (sigprof) [inline keyboards/nobreakfast/nb62/rules.mk:18]:
```suggestion
```
No need to mention VIA anywhere except in the `via` keymap.

File: keyboards/nobreakfast/nb62/rules.mk [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+# disable
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+NKRO_ENABLE = no		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+VIA_ENABLE = no 
+MOUSEKEY_ENABLE = no	# Mouse keys(+4700)
+CONSOLE_ENABLE = no	# Console for debug(+400)
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
```
Comment (sigprof) [inline keyboards/nobreakfast/nb62/rules.mk:22]:
```suggestion
```
This obsolete option should not be mentioned at all.

File: keyboards/nobreakfast/nb62/rules.mk [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# MCU name
+MCU = atmega32u4
+
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = yes	# Virtual DIP switch configuration(+1000)
```
Comment (sigprof) [inline keyboards/nobreakfast/nb62/rules.mk:9]: If you want to keep using `rules.mk`, please change the whole section to match https://github.com/qmk/qmk_firmware/blob/a239051c4a4779767059140892144dedea09aaf2/data/templates/avr/rules.mk (this is the last version of the `rules.mk` template before switching to the data-driven format as the default):
- fix the comment before the section, which is wrong in general (some options might be enabled by default and need an explicit `no` to disable);
- remove the obsolete `(+1000)`-like parts which refer to approximate code size (those values are often completely wrong);
- keep the order of options like in the template, add any extras below.

File: keyboards/nobreakfast/nb62/config.h [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+#endif
+
+/* BOOTMAGIC LITE */
+#define BOOTMAGIC_LITE_ROW 0
+#define BOOTMAGIC_LITE_COLUMN 1
+
+#endif
```
Comment (sigprof) [inline keyboards/nobreakfast/nb62/config.h:74]:
```suggestion
```

File: keyboards/nobreakfast/nb62/nb62.h [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef NB62_H
+#define NB62_H
```
Comment (sigprof) [inline keyboards/nobreakfast/nb62/nb62.h:19]:
```suggestion
#pragma once
```

File: keyboards/nobreakfast/nb62/nb62.h [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+	K400, K401, K402, K403,                               K409, K410, K411, K412, K413  \
+) { \
+	{ K000,  K001,  K002,  K003,  K004,  K005,  K006,  K007,  K008,  K009,  K010,  K011,  K012,  K013 }, \
+	{ K100,  K101,  K102,  K103,  K104,  K105,  K106,  K107,  K108,  K109,  K110,  K111,  K112,  K113 }, \
+	{ K200,  K201,  K202,  K203,  K204,  K205,  K206,  K207,  K208,  K209,  K210,  K211,  KC_NO, K213 }, \
+	{ K300,  K301,  K302,  K303,  K304,  K305,  KC_NO, K307,  K308,  K309,  K310,  K311,  KC_NO, K313 }, \
+	{ K400,  K401,  K402,  K403,  KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, K409,  K410,  K411,  K412,  K413 }  \
+}
+
+#endif
```
Comment (sigprof) [inline keyboards/nobreakfast/nb62/nb62.h:37]:
```suggestion
```

**Review by sigprof — CHANGES_REQUESTED** [review]
Unfortunately, your pull request does not have several important parts:

- The `info.json` file is required (as a minimum, it must contain the parts listed in the [PR Checklist](https://docs.qmk.fm/#/pr_checklist?id=keyboard-prs), but you may consider moving other things like matrix pins and layout definitions there too).
- The `readme.md` file is also required (again, see the PR Checklist).
- The `rules.mk` file should not contain obsolete comments with the approximate number of bytes; even better, the data that you have there should be moved to `info.json` instead (see https://github.com/qmk/qmk_firmware/tree/master/data/templates/keyboard for the template that is currently used by `qmk new-keyboard`).

File: keyboards/nobreakfast/nb62/config.h [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+#define RGBLED_NUM 62
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+#endif
+
+/* BOOTMAGIC LITE */
+#define BOOTMAGIC_LITE_ROW 0
+#define BOOTMAGIC_LITE_COLUMN 1
+
```
[the remark this answers is not in the document]
Comment (Nobreakfast) [reply keyboards/nobreakfast/nb62/config.h:73]: Thanks for your detailed advices and suggestions.

I am a noob in this qmk firmware, but want to build my own keyboard. I will fix my problem as soon as possible.

**Comment by Nobreakfast** [comment]
Sorry, this request does not follow the rule of fork in qmk, and I closed this request.

**Pull request closed without merging**